src/index.ts

```diff
@@ -113,22 +114,27 @@ export default fastifyPlugin<FastifyAutoroutesOptions>(
     }
 
     if (!fs.existsSync(dirPath)) {
-      return next(new Error(`${ERROR_LABEL} dir ${dirPath} does not exists`))
+      const CustomError = createError(
+        '1',
+        `${ERROR_LABEL} dir ${dirPath} must be a directory`
+      )
+      return next(new CustomError())
     }
 
     if (!fs.statSync(dirPath).isDirectory()) {
-      return next(
-        new Error(`${ERROR_LABEL} dir ${dirPath} must be a directory`)
+      const CustomError = createError(
+        '2',
+        `${ERROR_LABEL} dir ${dirPath} must be a directory`
       )
+      return next(new CustomError())
     }
 
-    let routes = await glob(`${dirPath}/**/[!._]*/[!._]*.{ts,js}`)
-    const routesModules: Record<string, StrictResource> = {}
-
     // glob returns ../../, but windows returns ..\..\
-    routes = routes.map((route) => path.normalize(route).replace(/\\/g, '/'))
     dirPath = path.normalize(dirPath).replace(/\\/g, '/')
```

Comment:
Replacing the path seperators here resolves the issue and glob returns the correct paths